src/main/groovy/org/shipkit/internal/gradle/versionupgrade/VersionUpgradeProducerPlugin.java

```diff
@@ -0,0 +1,125 @@
+package org.shipkit.internal.gradle.versionupgrade;
+
+import org.gradle.api.*;
+import org.gradle.api.tasks.Exec;
+import org.shipkit.gradle.configuration.ShipkitConfiguration;
+import org.shipkit.internal.gradle.configuration.DeferredConfiguration;
+import org.shipkit.internal.gradle.configuration.ShipkitConfigurationPlugin;
+import org.shipkit.internal.gradle.git.CloneGitRepositoryTask;
+import org.shipkit.internal.gradle.git.GitAuthPlugin;
+import org.shipkit.internal.gradle.util.TaskMaker;
+import org.shipkit.internal.util.ExposedForTesting;
+import org.shipkit.version.VersionInfo;
+
+import java.io.File;
+
+import static org.shipkit.internal.util.ArgumentValidation.notNull;
+
+/**
+ * BEWARE! This plugin is in incubating state, so its API may change in the future!
+ * The plugin applies following plugins:
+ *
+ * <ul>
+ *     <li>{@link ShipkitConfigurationPlugin}</li>
+ * </ul>
+ *
+ * and adds following tasks:
+ *
+ * <ul>
+ *     <li>cloneConsumerRepo{consumerRepository} - clones consumer repository into temporary directory</li>
+ *     <li>produceVersionUpgrade{consumerRepository} - runs task performVersionUpgrade on consumerRepository</li>
+ *     <li>produceVersionUpgrade - task aggregating all of the produceVersionUpgrade{consumerRepository} tasks</li>
```

Comment:
If you run it without "clean" it will fail on clone, since the temporary directory won't be empty. With clean if you run it and the version of project doesn't change it will fail on "git commit" since the file with dependency (build.gradle) will not change and there will be nothing to commit. It's won't be clear to the user what happened, so I'm thinking about checking if there was any actual change made, and configure subsequent tasks (git commit, push, create pull request) with .onlyIf condition based on it. What do you think about it?